enhancements/installer/cluster-api-based-installations.md

```diff
@@ -0,0 +1,347 @@
+be more actively developed than their Terraform counterparts. Furthermore, it will be necessary
+to ensure that the CAPI providers used by the Installer match the version of those in the payload.
+
+While this external dependency is a significant drawback, it is not unique to this design
+and is common throughout OpenShift (e.g. any time the API or library-go must be updated
+before being vendored into a component). To minimize the devex friction, we will focus
+on documenting a workflow for developing providers while working with the Installer.
+
+Additionally, we will explore designs to solve or mitigate these issues.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. UX design during install process as well as during failure (log collection). The Installer will dump
+(potentially prettified) controller logs. Once we reach a certain level of stability it may be worthwhile
+to implement a UI.
+
+2. Whether to use downstream OpenShift-specific `kube-apiserver` and `etcd` dependencies and how to source them?
+
+3. When should the Installer declare infrastructure provisioning failed? 
+
+### Test Plan
+
+As this is replacing existing functionality in the Installer, we can rely on existing
+testing infrastructure.
+
+### Graduation Criteria
+
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+#### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+As this enhancement only concerns the Installation process and affects only the underlying cluster
+infrastructure, this change should not affect existing cluster upgrades.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+During a failed install, the controller logs will contain useful information. The status of the CAPI manifests
+may also contain useful information,in which case it would be important to display that to users and collect
+for bugs and support cases. There is an open question about the best way to handle this UX, and we expect the answer to become more clear during development. 
+
+As the infrastructure will be reconciled by a controller, it will be possible to resolve issues during an ongoing
+installation, although this would not necessarily be a feature we would call attention to for documented use cases.
+
+Finally, the Installer will need to be able to identify when infrastructure provisioning has failed during an installation.
+Initially this will be achieved through a timeout. There is an open question about whether this can be done in a more
```

Comment:
If using a timeout, I suggest a soft timeout / warning since sometimes things go slowly for unpredictable but valid reasons. "Your installation is taking longer than expected. We are still waiting for X to be created. Use ctrl-C if you want to abort the installation."

Reply:
A hard timeout is at least consistent with what we're doing now. Let's put "changing timeout handling" on the list of non-goals for this proposal and tackle it as a separate topic.

Reply:
Added some more details in the open questions, to tackle in a later iteration